File: quickSort.go
package main

import "fmt"

func quickSort( list []int ,start int,end int){
	if start < end{
		i,j := start,end
		key := list[(start+end)/2]
		for i <= j{
			for list[i] < key{
				i++
			}
			for list[j] > key{
				j--
			}
			if i <=j{
				list[i],list[j] = list[j],list[i]
				i++
				j--
			}

		}
		if start < j{
			quickSort(list,start,j)
		}
		if end > i{
			quickSort(list,i,end)
		}

	}
}

func main(){
	list := []int{3,45,34,134,67,6,7657465,33}
	start := 0
	end := len(list)-1
	quickSort(list,start,end)
	fmt.Println(list)
}

File: quick_sort.py
# -*- coding: utf8 -*-
def quick_sort(my_list, start, end):
    # if start == end:
    #     return False
    if start < end:
        i, j = start, end
        base = my_list[i]
        while i < j:
            while (i < j) and (my_list[j] >= base):
                j = j - 1

            my_list[i] = my_list[j]

            while (i < j) and (my_list[i] <= base):
                i = i + 1

            my_list[j] = my_list[i]
        my_list[i] = base
        quick_sort(my_list, start, i - 1)
        quick_sort(my_list, j + 1, end)

    return my_list


myList = [49, 38, 65, 97, 76, 13, 27, 49]
print("Quick Sort: ")
quick_sort(myList, 0, len(myList) - 1)
print(myList)
